Gruntfile.js

```diff
@@ -176,33 +180,53 @@ module.exports = function( grunt ) {
 	} );
 
 	// Load grunt tasks from NPM packages
-	require( "load-grunt-tasks" )( grunt );
+	// Support: Node.js <4
+	// Don't load the eslint task in old Node.js, it won't parse.
+	require( "load-grunt-tasks" )( grunt, {
+		pattern: oldNode ? [ "grunt-*", "!grunt-eslint" ] : [ "grunt-*" ]
+	} );
 
 	// Integrate jQuery specific tasks
 	grunt.loadTasks( "build/tasks" );
 
-	grunt.registerTask( "lint", [ "jsonlint", "eslint:all" ] );
+	grunt.registerTask( "print_old_node_message", function() {
```

Comment:
I just don't see how we could do it using `grunt.task.run` without that many modifications as in the current version of this PR.

Anyway, I tried to encapsulate it as much as possible, in most cases it will require just unwrapping the task name from `runIfNewNode()` so no biggie.